ibis/expr/schema.py

```diff
@@ -133,6 +155,39 @@ def name_at_position(self, i):
             )
         return self.names[i]
 
+    def apply_to(self, df):
+        """Applies the Ibis schema to a pandas DataFrame
```

Comment:
Sorry what does this method do? Does it cast the types of df according to this ibis schema?

Reply:
As said in the previous comment, I just moved this method, since IMHO it was in the wrong file. And I finally ended up always loading the pandas backend, but if we eventually don't load it at `import ibis`, this change would be needed, since this code is needed even if the backend is not loaded.

But answering your question, my understanding is that when we convert the cursor returned by the backend to pandas, types are inferred by pandas. And this method takes the original schema of the ibis expression (or let's say the SQL table), and casts columns to have the same types.

Reply:
kk SGTM